From 7c2b6e1ebbc64f947a3e3f7dbbd40aa5454d4dac Mon Sep 17 00:00:00 2001
Message-Id: <7c2b6e1ebbc64f947a3e3f7dbbd40aa5454d4dac.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: "G, Pallavi" <pallavi.g@intel.com>
Date: Fri, 6 Sep 2013 15:45:39 +0530
Subject: [PATCH 023/312] FOR_UPSTREAM [VPG]: drm/i915: Reset lane as a
 disable sequence for VLV

As per application notes resetting lanes for Valleyview as part
of disable sequence. This patch is mandatory for BYT platform as
for BYT we should ensure to disable even lane during crtc disable
sequence.

(cherry picked from commit 44a0359d8f5787c6710a0cbd3e228bdaabe67719)

Issue: APDEV-1018
Change-Id: I87d132963c0cc2b2c4926306408613472169c396
Signed-off-by: Pallavi G <pallavi.g@intel.com>
Signed-off-by: Uma Shankar <uma.shankar@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index b5f413e..5256769 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5048,6 +5048,14 @@ static void i9xx_crtc_disable(struct drm_crtc *crtc)
 	intel_crtc->active = false;
 	intel_update_watermarks(crtc);
 
+	/*Reset lane for VLV platform*/
+	if (IS_VALLEYVIEW(dev)) {
+		if (intel_pipe_has_type(crtc, INTEL_OUTPUT_HDMI)) {
+			vlv_dpio_write(dev_priv, pipe, 0x8200, 0x00000000);
+			vlv_dpio_write(dev_priv, pipe, 0x8204, 0x00e00060);
+		}
+	}
+
 	mutex_lock(&dev->struct_mutex);
 	intel_update_fbc(dev);
 	intel_edp_psr_update(dev);
-- 
1.7.9.5

